integration/spark2/src/test/scala/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMapSuite.scala

```diff
@@ -413,6 +414,46 @@ class BloomCoarseGrainDataMapSuite extends QueryTest with BeforeAndAfterAll with
     checkQuery("fakeDm", shouldHit = false)
   }
 
+  test("test create datamaps on different column but hit only one") {
+    val originDistributedDatamapStatus = CarbonProperties.getInstance().getProperty(
+      CarbonCommonConstants.USE_DISTRIBUTED_DATAMAP,
+      CarbonCommonConstants.USE_DISTRIBUTED_DATAMAP_DEFAULT
+    )
+
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.USE_DISTRIBUTED_DATAMAP, "true")
+    val datamap1 = "datamap1"
+    val datamap2 = "datamap2"
+    sql(
+      s"""
+         | CREATE TABLE $bloomDMSampleTable(id INT, name STRING, city STRING, age INT)
+         | STORED BY 'carbondata'
+         |  """.stripMargin)
+    sql(
+      s"""
+         | CREATE DATAMAP $datamap1 ON TABLE $bloomDMSampleTable
+         | USING 'bloomfilter'
+         | DMProperties('INDEX_COLUMNS'='name', 'BLOOM_SIZE'='64000', 'BLOOM_FPP'='0.00001')
+      """.stripMargin)
+    sql(
+      s"""
+         | CREATE DATAMAP $datamap2 ON TABLE $bloomDMSampleTable
+         | USING 'bloomfilter'
+         | DMProperties('INDEX_COLUMNS'='city', 'BLOOM_SIZE'='64000', 'BLOOM_FPP'='0.00001')
+      """.stripMargin)
+
+    sql(
+      s"""
+         | INSERT INTO $bloomDMSampleTable
+         | VALUES(5,'a','beijing',21),(6,'b','shanghai',25),(7,'b','guangzhou',28)
+      """.stripMargin)
+    sql(s"SELECT * FROM $bloomDMSampleTable WHERE name='shanghai'").show()
```

Comment:
use expect to validate the result instead of show

Reply:
Is lucene datamap require similar fix? Please raise a JIRA issue to track it. Thanks

Reply:
Test shows that lucene is not affected by this. On the other hand, lucene gets wrong result if we apply this filter. Reason to be determined

Reply:
Test shows that lucene is not affected by this. On the other hand, lucene gets wrong result(Empty) if we apply this filter.(testname: "test lucene fine grain multiple data map on table" ).  Issue 2747 is raised to track